docs/design/queue-guarantee-resource-reservation-design.md

```diff
@@ -0,0 +1,100 @@
+# Volcano Resource Reservation For Queue
+
+@[qiankunli](https://github.com/qiankunli); Oct 11rd, 2021
+
+## Motivation
+As [issue 1101](https://github.com/volcano-sh/volcano/issues/1101) mentioned, Volcano should support resource reservation
+for specified queue. Requirement detail as follows:
+* Support reserving specified resources for specified queue
+* We only Consider non-preemption reservation.
+* Support enable and disable resource reservation for specified queue dynamically without restarting Volcano.
+* Support hard reservation resource specified and percentage reservation resource specified.
+
+@[Thor-wl](https://github.com/Thor-wl) already provide a design doc [Volcano Resource Reservation For Queue](https://github.com/volcano-sh/volcano/blob/master/docs/design/queue-resource-reservation-design.md)
+I do not implement all features above, supported feature are as follows:
+
+* Support reserving specified resources for specified queue
+* We only Consider non-preemption reservation.
+* Support enable and disable resource reservation for specified queue dynamically without restarting Volcano.
+* Support hard reservation resource specified
+
+## Consideration
+### Resource Request
+* The reserved resource cannot be more than the total resource amount of cluster at all dimensions.
+* If `capability` is set in a queue, the reserved resource must be no more than it at all dimensions.
+
+### Safety
+* Malicious application for large amount of resource reservation will cause jobs in other queue to block.
+
+## Design
+### API
+```
+apiVersion: scheduling.volcano.sh/v1beta1
+kind: Queue
+metadata:
+  name: q1
+spec:
+  reclaimable: true
+  weight: 1
+  guarantee:             // reservation key word
+    resource:            // specified reserving resource
+      cpu: 2c
+      memory: 4G
+```
+
+`guarantee.resource` list of reserving resource categories and amount.
+
+## Implementation
+
+保证 guarantee 机制，包含两种场景
+1. queue 配置没有变化时 保证guarantee 机制
+2. 新增一个带guarantee queue 或某个queue 的guarantee 变大时保证guarantee 机制
+
+### queue guarantee 配置没有变化时 保证guarantee 机制
+
+每一次开始调度之前，volcano 会为每个queue 计算好deserved 资源（代表了queue 本轮调度可以使用的资源），queue 中的task 要运行时，如果deserved > allocated, task 即可运行。为保证 guarantee 机制
+
+1. 任何时候，queue 的deserved 应大于 queue.guarantee。
+2. 如果有一个queue 设置了guarantee，相当于剩下的所有的queue 都设置了一个上限值，我们称之为realCapacity
+   a. 如果一个queue 未设置capacity，`realCapacity = 资源总量 - 其它queue的guarantee 之和`
```

Comment:
the "realCapaciy" a confusing word, let's get another name. like "limit" or some other words.